pkgs/applications/networking/instant-messengers/zoom-us/default.nix

```diff
@@ -66,15 +74,27 @@ in mkDerivation {
       runHook postInstall
     '';
 
-  postInstall = (makeDesktopItem {
-    name = "zoom-us";
-    exec = "$out/bin/zoom-us %U";
-    icon = "$out/share/zoom-us/application-x-zoom.png";
-    desktopName = "Zoom";
-    genericName = "Video Conference";
-    categories = "Network;Application;";
-    mimeType = "x-scheme-handler/zoommtg;";
-  }).buildCommand + ''
+  postInstall = ''
+    mkdir -p $out/share/{applications,appdata,icons}
+
+    # Desktop File
+    cp ${desktopIntegration}/us.zoom.Zoom.desktop $out/share/applications
+    substituteInPlace $out/share/applications/us.zoom.Zoom.desktop \
+        --replace "Exec=zoom" "Exec=$out/bin/zoom-us"
+
+    # Appdata
+    cp ${desktopIntegration}/us.zoom.Zoom.appdata.xml $out/share/appdata
+
+    # Icons
+    icon_sizes=("64" "96" "128" "256")
+
+    for icon in ''${icon_sizes[*]}
+    do
+        mkdir -p $out/share/icons/hicolor/$icon\x$icon/apps
```

Comment:
In general, should we use two-space indents in scripts, to match the Nix convention?

Reply:
I don't think so. IIRC this is a rampant problem.